samples/msal-angular-v2-samples/angular10-browser-sample/src/app/msal/msal.broadcast.service.ts

```diff
@@ -1,25 +1,39 @@
-import { Injectable } from '@angular/core';
+import { Inject, Injectable } from '@angular/core';
 import { Observable, Subject } from 'rxjs';
-import { MsalBroadcastEvent } from './constants';
-import { AuthenticationResult, AuthError } from '@azure/msal-browser';
-
-export interface MsalBroadcastMessage {
-  type: MsalBroadcastEvent;
-  payload: AuthenticationResult | AuthError;
-}
+import { InteractionType, MSAL_INSTANCE } from './constants';
+import { EventType, EventMessage, EventPayload, EventError, IPublicClientApplication } from '@azure/msal-browser';
 
 @Injectable()
 export class MsalBroadcastService {
   private _msalSubject: Subject<any>;
   public msalSubject$: Observable<any>;
 
-  constructor() {
-    this._msalSubject = new Subject<MsalBroadcastMessage>();
+  constructor(
+    @Inject(MSAL_INSTANCE) private msalInstance: IPublicClientApplication
+  ) {
+    this._msalSubject = new Subject<EventMessage>();
     this.msalSubject$  = this._msalSubject.asObservable();
+    this.msalInstance.addEventCallback((message: EventMessage) => {
+      this._msalSubject.next(message);
+    });
   }
 
-  broadcast(type: MsalBroadcastEvent, payload: AuthenticationResult | AuthError) {
-
-    this._msalSubject.next({type , payload});
+  /**
+   * Broadcasts events from msal-angular, same shape as events from msal-browser
+   * @param eventType 
+   * @param interactionType 
+   * @param payload 
+   * @param error 
+   */
+  broadcastAngularEvent(eventType: EventType, interactionType?: InteractionType, payload?: EventPayload, error?: EventError) {
```

Comment:
To consider: If we are broadcasting events from msal-angular that are not the events listed in the msal-browser eventType, we may have to create a new type for the eventType, eventMessage, eventPayload, etc.

Reply:
I think this function can be deleted since we won't be doing any Angular-specific broadcasting at this time.

Reply:
Makes sense. Removed.